src/daft-dsl/src/functions/kv/kv_functions.rs

```diff
@@ -0,0 +1,331 @@
+        let input_length = keys.len();
+        let result_series = Series::full_null(
+            "result",
+            &daft_core::datatypes::DataType::Binary,
+            input_length,
+        );
+        Ok(result_series)
+    }
+
+    fn get_return_field(
+        &self,
+        args: FunctionArgs<ExprRef>,
+        _schema: &Schema,
+    ) -> DaftResult<daft_core::datatypes::Field> {
+        let inputs = args.into_inner();
+        if inputs.len() != 2 {
+            return Err(DaftError::ValueError(
+                "kv_exists_with_name expects exactly 2 arguments".to_string(),
+            ));
+        }
+        Ok(daft_core::datatypes::Field::new(
+            "result",
+            daft_core::datatypes::DataType::Binary,
+        ))
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub struct KVPutWithStoreName;
+
+#[typetag::serde]
+impl ScalarUDF for KVPutWithStoreName {
+    fn name(&self) -> &'static str {
+        "kv_put_with_name"
+    }
+
+    fn call(&self, args: FunctionArgs<Series>) -> DaftResult<Series> {
+        let inputs = args.into_inner();
+        if inputs.len() != 3 {
+            return Err(DaftError::ValueError(
+                "kv_put_with_name expects exactly 3 arguments".to_string(),
+            ));
+        }
+        let key = &inputs[0];
+        let value = &inputs[1];
+        let store_name_series = &inputs[2];
+        if store_name_series.len() != 1 {
+            return Err(DaftError::ValueError(
+                "Store name series must contain exactly one element".to_string(),
+            ));
+        }
+        let name = store_name_series
+            .utf8()?
+            .get(0)
+            .ok_or_else(|| DaftError::ValueError("Missing store name".to_string()))?;
+        #[cfg(feature = "python")]
+        {
+            use daft_core::prelude::*;
+            // Build PySeries for key and value
+            let mut result_series_opt: Option<Series> = None;
+            Python::attach(|py| {
+                let daft_mod = py.import("daft.daft").unwrap();
+                let py_key = pyo3::Py::new(
+                    py,
+                    daft_core::python::PySeries {
+                        series: key.clone(),
+                    },
+                )
+                .unwrap();
+                let py_value = pyo3::Py::new(
+                    py,
+                    daft_core::python::PySeries {
+                        series: value.clone(),
+                    },
+                )
+                .unwrap();
+                let func = daft_mod.getattr("kv_put_direct_series").unwrap();
+                let py_res = func.call1((name, py_key, py_value)).unwrap();
+                let py_series: pyo3::PyRef<daft_core::python::PySeries> = py_res.extract().unwrap();
+                let mut s = py_series.series.clone();
+                s = s.rename("result");
+                result_series_opt = Some(s);
+            });
```

Comment:
properly handle errors here
```suggestion
    Python::attach(|py| {
                use pyo3::PyResult;

                let daft_mod = py.import("daft.daft")?;
                let py_key = pyo3::Py::new(
                    py,
                    daft_core::python::PySeries {
                        series: key.clone(),
                    },
                )?;
                let py_value = pyo3::Py::new(
                    py,
                    daft_core::python::PySeries {
                        series: value.clone(),
                    },
                )?;
                let func = daft_mod.getattr("kv_put_direct_series")?;
                let py_res = func.call1((name, py_key, py_value))?;
                let py_series: pyo3::PyRef<daft_core::python::PySeries> = py_res.extract()?;
                let mut s = py_series.series.clone();
                s = s.rename("result");
                result_series_opt = Some(s);
                PyResult::Ok(())
            })?;
```